frontend/src/lib/components/Editor.svelte

```diff
@@ -1,10 +1,52 @@
 <script lang="ts">
+	import { tick, onDestroy } from 'svelte';
 	import { currentFile } from '$lib/main';
 	import { addToast, ToastType } from '$lib/toast';
 	import { get } from 'svelte/store';
 	import { cache } from '$lib/cache';
 
-	export let saveChangesHandler: () => Promise<void>;
+	export let editorText: string;
+	export let previewWindow: HTMLElement;
+
+	let showCommitModal = false;
+	let commitModal: HTMLElement;
+	let commitMessageInput: HTMLInputElement;
+	let charCount = 500;
+
+	let previousFile: string | null = null;
+
+	function updateCharCount() {
+		charCount = 500 - commitMessageInput.value.length;
+		const charCountElement = document.getElementById('charCount');
+		if (charCountElement) {
+			charCountElement.textContent = `${charCount} characters remaining`;
```

Comment:
Svelte is reactive enough you shouldn't need to manually update this, just use `{charCount} characters remaining` where you're displaying it and svelte will automatically rerender the element when `charCount` changes.

This means that you can get rid of that ID. Generally, in svelte and a lot of other frameworks, you shouldn't ever need to find yourself using IDs much, there's often a more idiomatic way to do the same thing